### PR TITLE
Comments and Tests for sys_calls

### PR DESCRIPTION
**Pull request by Anway-Agte**
## Description

Fixes # (issue)
The following changes include the tests and comments in the code for the "get_pid" and "get_ppid" system call under RustPosix.

 

### Type of change

- [ ] This change adds tests and comments to existing code

## How Has This Been Tested?

 

All the tests are inside the ```lind_project/src/safeposix-rust/src/tests/sys_tests/rs``` 

- Test A - `ut_lind_getpid()`
- Test B - `lut_lind_get_ppid()`

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, lind-project)

### CODE REVIEW

File: src/tests/sys_tests.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+pub mod sys_tests {
+    use super::super::*;
+    use crate::interface;
+    use crate::safeposix::syscalls::sys_calls::*;
+    use crate::safeposix::{cage::*, dispatcher::*, filesystem};
+    use libc::c_void;
+    use std::fs::OpenOptions;
+    use std::os::unix::fs::PermissionsExt; 
+
+    pub fn test_fs() {
```
Comment (rennergade) [inline src/tests/sys_tests.rs]: this should technically be "test_sys" to match the other files syntax

File: src/safeposix/syscalls/sys_calls.rs [hunk 2]
```diff
@@ -38,6 +38,13 @@ impl Cage {
         }
     }
 
+
```
Comment (rennergade) [inline src/safeposix/syscalls/sys_calls.rs]: All of these headers need to be changed to match the RustDoc format. Checkout [this PR](https://github.com/Lind-Project/safeposix-rust/pull/266/files#diff-588741f51e18dec64c5a7299f2caba89320fafeac513840046d6cf3b87064c8b)

and look at the header comments for mkdir and fcntl

**Review by rennergade — CHANGES_REQUESTED** [review]
Besides the header comments being updated these functions generally need much more comments fleshing out why we do certain steps. Look at the recent PRs that we've merged for examples of what were looking for!

File: src/safeposix/syscalls/sys_calls.rs (hunk 2 above)
Comment (rennergade) [reply src/safeposix/syscalls/sys_calls.rs]: remove this extra line for this pr

File: src/safeposix/syscalls/sys_calls.rs [hunk 3]
```diff
@@ -221,6 +222,7 @@ impl Cage {
             new_semtable.insert((*pair.key()).clone(), pair.value().clone());
         }
 
+        // Create a new cage object for the child process
```
Comment (rennergade) [inline src/safeposix/syscalls/sys_calls.rs]: remove this comment for this pr

File: src/safeposix/syscalls/sys_calls.rs [hunk 4]
```diff
@@ -425,12 +453,21 @@ impl Cage {
...
+    /// The real user id is the user who invoked the calling process. 
+    /// As Lind only allows one user, a default value is returned. 
+    /// 
+    /// ### Arguments
+    ///  
+    /// The `getuid_syscall` does not take any arguments
+    /// 
+    /// ### Returns
+    /// 
+    /// Returns a 32 bit default integer (or -1) representing the user
```
Comment (JustinCappos) [inline src/safeposix/syscalls/sys_calls.rs:472]: But what we do here is different.  We always return a default number

File: src/safeposix/syscalls/sys_calls.rs [hunk 5]
```diff
@@ -411,11 +430,20 @@ impl Cage {
...
+    /// The `getegid_syscall` returns the effective group id of the user who invoked the process.
+    /// Since Lind is only run in one group a default value (or -1) is returned. 
+    /// 
+    /// ### Arguments 
+    /// 
+    /// This syscall does not take any arguments
+    /// 
+    /// ### Returns
+    /// 
+    /// Returns a 32 bit integer value which represents the effective group
```
Comment (JustinCappos) [inline src/safeposix/syscalls/sys_calls.rs]: Right, but what do we do?

File: src/tests/sys_tests.rs [hunk 6]
```diff
@@ -6,15 +6,10 @@ pub mod sys_tests {
...
-    use crate::safeposix::{cage::*, dispatcher::*, filesystem};
-    use libc::c_void;
-    use std::fs::OpenOptions;
-    use std::os::unix::fs::PermissionsExt; 
 
-    pub fn test_fs() {
+    pub fn tests_sys() {
         ut_lind_getpid(); 
         ut_lind_getppid(); 
-        ut_lind_getpid();
```
Comment (JustinCappos) [inline src/tests/sys_tests.rs]: Are we intending these changes?

**Review by JustinCappos — CHANGES_REQUESTED** [review]
A few minor comments, but I'll mark are request changes because I don't understand the test difference.

**Comment by rennergade** [comment]
I think this is a good start. @davidge20 can you give this a review?

File: src/safeposix/syscalls/sys_calls.rs [hunk 7]
```diff
@@ -370,41 +370,131 @@ impl Cage {
...
+    /// 
+    /// ### Arguments
+    ///  
+    /// The `getuid_syscall` does not take any arguments
+    /// 
+    /// ### Returns
+    /// 
+    /// Returns a 32 bit default integer (or -1) representing the user
     pub fn getuid_syscall(&self) -> i32 {
+        // The call returns -1 to indicate that the call happened during the loading stage
```
Comment (rennergade) [inline src/safeposix/syscalls/sys_calls.rs]: I think a more precise comment here would be:

"We return -1 for the first call for compatibility with the dynamic loader. For subsequent calls we return our default value."

**Review by rennergade — APPROVED** [review]
Looks good! Approved!

**Pull request merged**